src/webview/js/js.js

```diff
@@ -904,6 +904,28 @@ documentBody.addEventListener('click', (e: MouseEvent) => {
     return;
   }
 
+  if (target.matches('.poll-vote')) {
+    const messageElement = target.closest('.message');
+    if (messageElement) {
+      // This duplicates some logic from PollData.handle.vote.outbound, but it's
+      // much simpler to just duplicate it than it is to thread a callback all
+      // the way over here.
+      const current_vote = requireAttribute(target, 'data-voted') === 'true';
+      const vote = current_vote ? -1 : 1;
+      sendMessage({
+        type: 'vote',
+        messageId: requireNumericAttribute(messageElement, 'data-msg-id'),
+        key: requireAttribute(target, 'data-key'),
+        vote,
+      });
+      target.setAttribute('data-voted', (!current_vote).toString());
+      target.innerText = (parseInt(target.innerText, 10) + vote).toString();
+    } else {
+      throw new Error('Message element not found');
```

Comment:
Let's put the error case first, so it's immediately next to the condition that triggers it.

That also means the normal happy case doesn't get indented, and can be read just linearly down the screen.

Reply:
Good call, done.